base/MAPL_Resource.F90

```diff
@@ -0,0 +1,343 @@
+#include "MAPL_Exceptions.h"
+#include "MAPL_ErrLog.h"
+#include "unused_dummy.H"
+
+!=============================================================================
+!FPP macros for repeated (type-dependent) code
+
+#ifdef SET_VAL
+#  undef SET_VAL
+#endif
+
+#define SET_VAL(T) \
```

Comment:
Not strictly necessary, but maybe it would be best to put `val` as an argument to the macro?

I'm also wondering if `GET` might be more indicative than `SET`, but I can see both arguments.   The procedure is "getting" values but the macro is "setting".  But it is also getting from ESMF ...

Give it some thought, but don't spend time on that.

Reply:
I will include `val` to make it more funtionish. I'll keep `Set` since the macro ends up setting `val` locally for return.

Reply:
Done.